### ISSUE TITLE
IndexError fetching segmentation data

### ISSUE THREAD

**Issue by schlegelp** [seung-lab/cloud-volume#294]
Hi. There is probably something very obvious that I'm doing wrong but I keep getting index errors trying to access certain indices:

```Python
>>> from cloudvolume import CloudVolume
>>> ts = CloudVolume('precomputed://gs://fafb-ffn1-20190805/tissue_mask_argmax', mip=6
>>> ts[0,100,0] 
--------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-67-366f884d101e> in <module>
----> 1 ts[0,100,0]

~/.pyenv/versions/3.7.5/lib/python3.7/site-packages/cloudvolume/frontends/precomputed.py in __getitem__(self, slices)
    519     requested_bbox = Bbox.from_slices(slices)
    520 
--> 521     img = self.download(requested_bbox, self.mip)
    522     return img[::steps.x, ::steps.y, ::steps.z, channel_slice]
    523 

~/.pyenv/versions/3.7.5/lib/python3.7/site-packages/cloudvolume/frontends/precomputed.py in download(self, bbox, mip, parallel, segids, preserve_zeros)
    555       parallel = self.parallel
    556 
--> 557     img = self.image.download(bbox, mip, parallel=parallel)
    558 
    559     if segids is None:                                                                              | 0/1 [00:00<?, ?it/s]

~/.pyenv/versions/3.7.5/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/image/__init__.py in download(self, bbox, mip, parallel, location, retain, use_shared_memory, use_file, order)
    120         progress=self.config.progress,
    121         fill_missing=self.fill_missing,
--> 122         order=order,
    123       )
    124     else:

~/.pyenv/versions/3.7.5/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/image/rx.py in download_sharded(requested_bbox, mip, meta, cache, spec, compress, progress, fill_missing, order)
     63   for gridpoint in gpts:
     64     zcurve_code = compressed_morton_code(gridpoint, grid_size)
---> 65     chunkdata = reader.get_data(zcurve_code, meta.key(mip))
     66 
     67     cutout_bbox = Bbox(

~/.pyenv/versions/3.7.5/lib/python3.7/site-packages/cloudvolume/datasource/precomputed/sharding.py in get_data(self, label, path)
    237       minishard_index[i, 1] += minishard_index[i-1, 1] + minishard_index[i-1, 2]
    238 
--> 239     idx = np.where(minishard_index == label)[0][0]
    240     _, offset, size = minishard_index[idx,:]
    241     offset = int(offset + index_offset)

IndexError: index 0 is out of bounds for axis 0 with size 0
```

Running cloudvolume `0.65.5`. Any ideas?

**Comment by william-silversmith** [comment]
Hmm this looks like a problem with either the data or my implementation of
the sharded format reader. I'll have to take a closer look. Don't worry,
this isn't your fault. ☺️

**Comment by william-silversmith** [comment]
Hi Philipp, 

I looked into this issue and I found a few issues with the implementation. Neuroglancer appears to be able to [read this data](https://neuromancer-seung-import.appspot.com/#!%7B%22layers%22:%5B%7B%22source%22:%22precomputed://gs://fafb-ffn1-20190805/tissue_mask_argmax%22%2C%22crossSectionRenderScale%22:1201.6235110447383%2C%22type%22:%22segmentation%22%2C%22selectedAlpha%22:0.51%2C%22notSelectedAlpha%22:0.07%2C%22skeletonRendering%22:%7B%22mode2d%22:%22lines_and_points%22%2C%22mode3d%22:%22lines%22%7D%2C%22name%22:%22tissue_mask_argmax%22%7D%5D%2C%22navigation%22:%7B%22pose%22:%7B%22position%22:%7B%22voxelSize%22:%5B16%2C16%2C40%5D%2C%22voxelCoordinates%22:%5B41437%2C15849%2C3260%5D%7D%7D%2C%22zoomFactor%22:321.3685907710016%7D%2C%22selectedLayer%22:%7B%22layer%22:%22tissue_mask_argmax%22%7D%2C%22layout%22:%22xy%22%7D). So the problem has to lie with CV. 

The problems I've fixed are in #296 and #298 but I'm still seeing problems with mip 6 where some sections come up black. I've also been able to segfault the `compressed-segmentation` module by doing this:

```
ts = CloudVolume('precomputed://gs://fafb-ffn1-20190805/tissue_mask_argmax', 
  progress=True, fill_missing=True, mip=6, autocrop=True, cache=True)
img = ts[:,:,50:100].astype(np.uint8)
```

I'll release the work I have so far since it will allow you to at least do some reading, but it seems there's more to be done.

**Comment by william-silversmith** [comment]
CloudVolume 0.66.0 is out and contains the aforementioned fixes, though there are still problems.

**Comment by william-silversmith** [comment]
I'm assuming this problem is fixed too now in #404 #410
